Bottom: 03147211e4c277726edc0f1afb8f0cde7fe7b6f2
Top:    aa52e752996b1cbfa7fde9248884b285be9378c8
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-10 14:07:09 -0500

Add buildbot PTXdist worker Docker image


---
diff --git a/docker/buildbot-worker-ptxdist/Dockerfile b/docker/buildbot-worker-ptxdist/Dockerfile
new file mode 100644
index 00000000..ee6bfeb5
--- /dev/null
+++ b/docker/buildbot-worker-ptxdist/Dockerfile
@@ -0,0 +1,93 @@
+FROM        krallin/ubuntu-tini:trusty
+
+ENV PTXDIST_VERSION 2012.09.1
+
+WORKDIR /buildbot
+
+RUN apt-get update \
+&&  DEBIAN_FRONTEND=noninteractive \
+    apt-get -y install -q wget \
+&&  wget -qO /etc/apt/trusted.gpg.d/ptx-archive-key.gpg https://debian.pengutronix.de/debian/ptx-archive-key.gpg \
+&&  wget -qO /etc/apt/sources.list.d/pengutronix.list   https://debian.pengutronix.de/debian/pengutronix.list \
+&&  apt-get update \
+&&  DEBIAN_FRONTEND=noninteractive \
+    apt-get -y install -q \
+        oselas.toolchain-2011.11.3-armeb-xscale-linux-gnueabi-gcc-4.6.2-glibc-2.14.1-binutils-2.21.1a-kernel-2.6.39-sanitized \
+        oselas.toolchain-2012.12.1-arm-cortexa8-linux-gnueabi-gcc-4.7.3-glibc-2.16.0-binutils-2.22-kernel-3.6-sanitized \
+        oselas.toolchain-2012.12.1-i686-atom-linux-gnu-gcc-4.7.2-glibc-2.16.0-binutils-2.22-kernel-3.6-sanitized \
+&&  rm -rf /var/lib/apt/lists/*
+
+# Install security updates and required packages
+RUN apt-get update \
+&&  DEBIAN_FRONTEND=noninteractive \
+    apt-get -y install -q \
+        autoconf \
+        bison \
+        build-essential \
+        ccache \
+        comerr-dev \
+        default-jre-headless \
+        docbook-xml \
+        docbook-xsl \
+        groff-base \
+        libgdk-pixbuf2.0-dev \
+        libgtk2.0-bin \
+        libicu-dev \
+        libncurses5 \
+        libncurses5-dev \
+        libswitch-perl \
+        libxml-simple-perl \
+        libxml2-utils \
+        lzop \
+        flex \
+        gawk \
+        gcc \
+        gconf2 \
+        gettext \
+        git \
+        libtool \
+        make \
+        python-dev \
+        python-libxml2 \
+        python-tdb \
+        ruby \
+        subversion \
+        ss-dev \
+        texinfo \
+        unzip \
+        wget \
+        xsltproc \
+        yasm \
+&&  rm -rf /var/lib/apt/lists/*
+
+ARG BUILDBOT_VERSION
+# Install required python packages, and twisted
+RUN wget https://bootstrap.pypa.io/get-pip.py \
+&&  python get-pip.py --no-cache-dir \
+&&  pip --no-cache-dir install \
+        'virtualenv' \
+        'twisted[tls]' \
+        buildbot-worker${BUILDBOT_VERSION:+==${BUILDBOT_VERSION}} \
+&&  rm -rf get-pip.py
+
+ARG BUILDBOT_UID=1000
+COPY buildbot/ /home/buildbot/
+RUN useradd --comment "Buildbot Server" --home-dir "/home/buildbot" --shell "/bin/bash" --uid ${BUILDBOT_UID} --user-group buildbot \
+&&  chown -v -R buildbot:buildbot "/buildbot" \
+&&  chown -v -R buildbot:buildbot "/home/buildbot" \
+&&  useradd --comment "Gnome Display Manager" --home-dir "/var/lib/gdm" --shell "/sbin/nologin" --user-group --system gdm
+
+# Install ptxdist, a build system
+COPY src /usr/src/ptxdist-${PTXDIST_VERSION}
+RUN cd /usr/src/ptxdist-${PTXDIST_VERSION} \
+&&  touch ptxdist \
+&&  echo "${PTXDIST_VERSION}" > .tarball-version \
+&&  ./autogen.sh \
+&&  ./configure \
+&&  make \
+&&  make install \
+&&  make distclean
+
+USER buildbot
+
+CMD ["/home/buildbot/start.sh"]
diff --git a/docker/buildbot-worker-ptxdist/buildbot/.ccache/ccache.conf b/docker/buildbot-worker-ptxdist/buildbot/.ccache/ccache.conf
new file mode 100644
index 00000000..150374b8
--- /dev/null
+++ b/docker/buildbot-worker-ptxdist/buildbot/.ccache/ccache.conf
@@ -0,0 +1,8 @@
+# Set maximum cache size to 10 GB:
+max_size = 10G
+
+# Disable hard linking for shared cache
+hard_link = false
+
+# Place cache directory in common location
+cache_dir = /cache/ccache
diff --git a/docker/buildbot-worker-ptxdist/buildbot/.gitconfig b/docker/buildbot-worker-ptxdist/buildbot/.gitconfig
new file mode 100644
index 00000000..efd988f4
--- /dev/null
+++ b/docker/buildbot-worker-ptxdist/buildbot/.gitconfig
@@ -0,0 +1,3 @@
+[user]
+	name = BuildBot
+	email = buildbot@novatechweb.com
diff --git a/docker/buildbot-worker-ptxdist/buildbot/.ptxdist/ptxdistrc-git b/docker/buildbot-worker-ptxdist/buildbot/.ptxdist/ptxdistrc-git
new file mode 100644
index 00000000..7dbac6ef
--- /dev/null
+++ b/docker/buildbot-worker-ptxdist/buildbot/.ptxdist/ptxdistrc-git
@@ -0,0 +1,75 @@
+#
+# Automatically generated file; DO NOT EDIT.
+# PTXdist 2012.09.1-00373-gee9622f
+#
+
+#
+# PTXDIST Setup Menu
+#
+
+#
+# User
+#
+PTXCONF_SETUP_USER_NAME="NovaTech Buildbot"
+PTXCONF_SETUP_USER_EMAIL="buildbot@novatechweb.com"
+
+#
+# Proxies
+#
+PTXCONF_SETUP_FTP_PROXY=""
+PTXCONF_SETUP_HTTP_PROXY=""
+PTXCONF_SETUP_HTTPS_PROXY=""
+PTXCONF_SETUP_NO_PROXY=""
+
+#
+# Project Searchpath
+#
+PTXCONF_SETUP_PROJECTPATH="${PTXDIST_TOPDIR}/projects"
+
+#
+# Source Directories
+#
+PTXCONF_SETUP_SRCDIR="/cache/downloads"
+
+#
+# Source Download
+#
+# PTXCONF_SETUP_NO_DOWNLOAD is not set
+# PTXCONF_SETUP_PTXMIRROR_ONLY is not set
+PTXCONF_SETUP_PTXMIRROR="http://www.pengutronix.de/software/ptxdist/temporary-src"
+PTXCONF_SETUP_DEBMIRROR="http://ftp.uni-kl.de/debian"
+PTXCONF_SETUP_SFMIRROR="http://downloads.sourceforge.net/sourceforge"
+PTXCONF_SETUP_GNUMIRROR="http://ftp.uni-kl.de/pub/gnu"
+PTXCONF_SETUP_XORGMIRROR="http://xorg.mirrors.pair.com/pub/X11/ftp.x.org"
+PTXCONF_SETUP_KERNELMIRROR="http://www.kernel.org/pub/linux"
+# PTXCONF_SETUP_CHECK_ALWAYS is not set
+PTXCONF_SETUP_CHECK_NOTEMPTY=y
+# PTXCONF_SETUP_CHECK_NEVER is not set
+PTXCONF_SETUP_CHECK="notempty"
+
+#
+# IPKG Repository
+#
+PTXCONF_SETUP_IPKG_REPOSITORY="/cache/ipkg-repository"
+
+#
+# Java SDK
+#
+PTXCONF_SETUP_JAVA_SDK="/usr"
+
+#
+# Developer Options
+#
+# PTXCONF_SETUP_DISABLE_LOCAL_CHECK is not set
+PTXCONF_SETUP_ENV_WHITELIST=""
+# PTXCONF_SETUP_COMMON_CACHE is not set
+# PTXCONF_SETUP_GEN_DEP_TREE is not set
+# PTXCONF_SETUP_CHECK_EXIT_ON_ERROR is not set
+PTXCONF_SETUP_CCACHE=y
+PTXCONF_SETUP_PATCHIN_GIT=y
+# PTXCONF_SETUP_NFS_REL_SYMLINK is not set
+# PTXCONF_SETUP_DIRECT_CLEAN is not set
+PTXCONF_SETUP_HOST_CPP="cpp"
+PTXCONF_SETUP_HOST_CC="gcc"
+PTXCONF_SETUP_HOST_CXX="g++"
+PTXCONF_SETUP_HOST_MAKE="make"
diff --git a/docker/buildbot-worker-ptxdist/buildbot/.pydistutils.cfg b/docker/buildbot-worker-ptxdist/buildbot/.pydistutils.cfg
new file mode 100644
index 00000000..8dc2a29b
--- /dev/null
+++ b/docker/buildbot-worker-ptxdist/buildbot/.pydistutils.cfg
@@ -0,0 +1,2 @@
+[easy_install]
+index-url=http://172.16.103.72/repository/PyPi_Proxy/simple
diff --git a/docker/buildbot-worker-ptxdist/buildbot/buildbot.tac b/docker/buildbot-worker-ptxdist/buildbot/buildbot.tac
new file mode 100644
index 00000000..37012ab4
--- /dev/null
+++ b/docker/buildbot-worker-ptxdist/buildbot/buildbot.tac
@@ -0,0 +1,38 @@
+import fnmatch
+import os
+import sys
+
+from twisted.application import service
+from twisted.python.log import FileLogObserver
+from twisted.python.log import ILogObserver
+
+from buildbot_worker.bot import Worker
+
+# setup worker
+basedir = os.path.abspath(os.path.dirname(__file__))
+application = service.Application('buildbot-worker')
+
+
+application.setComponent(ILogObserver, FileLogObserver(sys.stdout).emit)
+# and worker on the same process!
+buildmaster_host = os.environ.get("BUILDBOT_MASTER", 'localhost')
+port = int(os.environ.get("BUILDBOT_WORKER_PORT", 9989))
+workername = os.environ.get("BUILDBOT_WORKER_NAME", 'docker')
+passwd = os.environ.get("BUILDBOT_WORKER_PASS")
+
+# delete the password from the environ so that it is not leaked in the log
+blacklist = os.environ.get("BUILDBOT_ENVIRONMENT_BLACKLIST", "BUILDBOT_WORKER_PASSWORD").split(',')
+for name in list(os.environ.keys()):
+    for toremove in blacklist:
+        if fnmatch.fnmatch(name, toremove):
+            del os.environ[name]
+
+keepalive = 600
+umask = None
+maxdelay = 300
+allow_shutdown = None
+
+s = Worker(buildmaster_host, port, workername, passwd, basedir,
+           keepalive, umask=umask, maxdelay=maxdelay,
+           allow_shutdown=allow_shutdown)
+s.setServiceParent(application)
diff --git a/docker/buildbot-worker-ptxdist/buildbot/start.sh b/docker/buildbot-worker-ptxdist/buildbot/start.sh
new file mode 100755
index 00000000..a6343a88
--- /dev/null
+++ b/docker/buildbot-worker-ptxdist/buildbot/start.sh
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+SSH_BITS=${SSH_BITS-"2048"}
+SSH_TYPE=${SSH_TYPE-"rsa"}
+SSH_FILE=${SSH_FILE-"$HOME/.ssh/id_rsa"}
+
+BUILDBOT_DATA=${BUILDBOT_DATA-"/buildbot"}
+
+ssh-keygen -q -b ${SSH_BITS} -t ${SSH_TYPE} -N '' -f ${SSH_FILE} 0>&- 1>/dev/null 2>/dev/null
+
+ln -v -s -f ${HOME}/buildbot.tac $BUILDBOT_DATA
+exec twistd --nodaemon --logfile=- --pidfile=/tmp/twistd.pid --python=buildbot.tac
